src/Lang/Cpp/CppDocGen.ts

```diff
@@ -228,6 +247,44 @@ export class CppDocGen implements IDocGen {
         }
     }
 
+    protected generateVersionTag(lines: string[]) {
+        if (this.cfg.File.versionTag.trim().length !== 0) {
+            lines.push(this.cfg.C.commentPrefix + this.cfg.File.versionTag);
+        }
+    }
+
+    protected generateCopyrightTag(lines: string[]) {
+        // This currently only supports year substitution
+        this.cfg.File.copyrightTag.forEach((element) => {
+            this.generateFromTemplate(
+                lines,
+                this.cfg.yearTemplateReplace,
+                element,
+                [moment().format("YYYY")],
+            );
+        });
+    }
+
+    protected generateCustomTag(lines: string[]) {
+        let dateFormat: string = "YYYY-MM-DD"; // Default to ISO standard if not defined
+        if ( this.cfg.Generic.dateFormat.trim().length !== 0) {
+            dateFormat = this.cfg.Generic.dateFormat; // Overwrite with user format
+        }
+        // For each line of the customTag
+        this.cfg.File.customTag.forEach((element) => {
+            // Allow any of date, year, author, email to be replaced
```

Comment:
This should also be in the description of the custom tag for the settings file.
Maybe some sort of list of which replacements are supported for the custom block.